### PR TITLE
[M] CANDLEPIN-528: Update problematic terms for candlepin

### PR DESCRIPTION
**Pull request by bakajstep**
- Rename problematics words

### CODE REVIEW

File: src/main/java/org/candlepin/policy/js/pool/PoolRules.java [hunk 1]
```diff
@@ -225,11 +225,11 @@ private Pool createBonusPool(Pool masterPool, List<Pool> existingPools) {
         return null;
     }
 
-    private boolean hasMasterPool(List<Pool> pools) {
+    private boolean hasPrimaryPool(List<Pool> pools) {
         if (pools != null) {
             for (Pool pool : pools) {
                 SourceSubscription srcSub = pool.getSourceSubscription();
-                if (srcSub != null && "master".equals(srcSub.getSubscriptionSubKey())) {
+                if (srcSub != null && "primary".equals(srcSub.getSubscriptionSubKey())) {
```
Comment (Ceiu) [inline src/main/java/org/candlepin/policy/js/pool/PoolRules.java:232]: value change

File: src/main/java/org/candlepin/policy/js/pool/PoolRules.java [hunk 2]
```diff
@@ -312,39 +312,39 @@ public List<PoolUpdate> updatePools(List<Pool> floatingPools, Map<String, Produc
...
         for (Pool existingPool : existingPools) {
             log.debug("Checking pool: {}", existingPool);
 
-            // Ensure subscription details are maintained on the master pool
-            if ("master".equalsIgnoreCase(existingPool.getSubscriptionSubKey())) {
-                existingPool.setUpstreamPoolId(masterPool.getUpstreamPoolId());
-                existingPool.setUpstreamEntitlementId(masterPool.getUpstreamEntitlementId());
-                existingPool.setUpstreamConsumerId(masterPool.getUpstreamConsumerId());
+            // Ensure subscription details are maintained on the primary pool
+            if ("primary".equalsIgnoreCase(existingPool.getSubscriptionSubKey())) {
```
Comment (Ceiu) [inline src/main/java/org/candlepin/policy/js/pool/PoolRules.java:332]: value change

File: src/test/java/org/candlepin/controller/EntitlementCertificateGeneratorTest.java [hunk 3]
```diff
@@ -353,7 +353,7 @@ public void testNonLazyRegenerate() throws Exception {
         Product product = TestUtil.createProduct();
 
         Pool pool = TestUtil.createPool(owner, product);
-        pool.setSourceSubscription(new SourceSubscription("source-sub-id", "master"));
+        pool.setSourceSubscription(new SourceSubscription("source-sub-id", "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/controller/EntitlementCertificateGeneratorTest.java:356]: value change

File: src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java [hunk 4]
```diff
@@ -1330,12 +1330,12 @@ public void testRefreshPoolsWithRemovedMasterPool() {
         List<Pool> pools = poolCurator.getBySubscriptionId(owner, sub.getId());
         assertEquals(2, pools.size());
         String bonusId =  "";
-        String masterId = "";
+        String primaryId = "";
 
         for (Pool p : pools) {
-            if (p.getSourceSubscription().getSubscriptionSubKey().equals("master")) {
+            if (p.getSourceSubscription().getSubscriptionSubKey().equals("primary")) {
```
Comment (Ceiu) [inline src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java:1336]: value change

File: src/test/java/org/candlepin/controller/PoolManagerTest.java [hunk 5]
```diff
@@ -445,7 +445,7 @@ public void testRefreshPoolsOnlyRegeneratesFloatingWhenNecessary() {
 
         // Should be unchanged
         Pool p = TestUtil.createPool(product);
-        p.setSourceSubscription(new SourceSubscription(sub.getId(), "master"));
+        p.setSourceSubscription(new SourceSubscription(sub.getId(), "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/controller/PoolManagerTest.java:448]: value change

File: src/test/java/org/candlepin/controller/RefresherTest.java [hunk 6]
```diff
@@ -111,7 +111,7 @@ public void testPoolOnlyExaminedOnceProductAndOwner() {
         product.setUuid("product uuid");
 
         Pool pool = new Pool();
-        pool.setSourceSubscription(new SourceSubscription("subId", "master"));
+        pool.setSourceSubscription(new SourceSubscription("subId", "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/controller/RefresherTest.java:114]: value change

File: src/test/java/org/candlepin/controller/RefresherTest.java [hunk 7]
```diff
@@ -144,7 +144,7 @@ public void testPoolOnlyExaminedOnceTwoProducts() {
         product2.setUuid("product id 2");
 
         Pool pool = new Pool();
-        pool.setSourceSubscription(new SourceSubscription("subId", "master"));
+        pool.setSourceSubscription(new SourceSubscription("subId", "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/controller/RefresherTest.java:147]: value change

File: src/test/java/org/candlepin/model/PoolCuratorTest.java [hunk 8]
```diff
@@ -1700,13 +1702,13 @@ public void testGetPoolsOrderedByProductNameAscending() {
         Owner owner1 = this.createOwner();
 
         Pool p1 = TestUtil.createPool(owner1, this.generateProduct(owner1, "p1", "xyz"))
-            .setSourceSubscription(new SourceSubscription("subscriptionId-phil", "master"));
+            .setSourceSubscription(new SourceSubscription("subscriptionId-phil", "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/model/PoolCuratorTest.java:1705]: value change (several others throughout this file; omitting additional comments)

File: src/test/java/org/candlepin/policy/AutobindRulesTest.java [hunk 9]
```diff
@@ -3497,7 +3497,7 @@ public void testFindBestWillNotCompleteAPartialStackFromAnotherId() {
     protected Pool createPool(Owner owner, Product product, int quantity, Date startDate, Date endDate) {
         Pool p = TestUtil.createPool(owner, product, quantity);
         p.setId("testpool" + TestUtil.randomInt());
-        p.setSourceSubscription(new SourceSubscription("testsub" + TestUtil.randomInt(), "master"));
+        p.setSourceSubscription(new SourceSubscription("testsub" + TestUtil.randomInt(), "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/policy/AutobindRulesTest.java:3500]: value change

File: src/test/java/org/candlepin/sync/SubscriptionReconcilerTest.java [hunk 10]
```diff
@@ -139,7 +139,7 @@ public Pool convertFromDTO(SubscriptionDTO sub) {
             .setUpstreamEntitlementId(sub.getUpstreamEntitlementId());
 
         if (sub.getId() != null) {
-            pool.setSourceSubscription(new SourceSubscription(sub.getId(), "master"));
+            pool.setSourceSubscription(new SourceSubscription(sub.getId(), "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/sync/SubscriptionReconcilerTest.java:142]: value change

File: src/test/java/org/candlepin/test/DatabaseTestFixture.java [hunk 11]
```diff
@@ -605,7 +605,7 @@ protected Pool createPool(Owner owner, Product product, Long quantity, Date star
             .setContractNumber(DEFAULT_CONTRACT)
             .setAccountNumber(DEFAULT_ACCOUNT)
             .setOrderNumber(DEFAULT_ORDER)
-            .setSourceSubscription(new SourceSubscription(Util.generateDbUUID(), "master"));
+            .setSourceSubscription(new SourceSubscription(Util.generateDbUUID(), "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/test/DatabaseTestFixture.java:608]: value change

File: src/test/java/org/candlepin/test/TestUtil.java [hunk 12]
```diff
@@ -570,7 +570,7 @@ public static Pool createPool(Owner owner, Product product, int quantity) {
             .setContractNumber("SUB234598S" + random)
             .setAccountNumber("ACC123" + random)
             .setOrderNumber("ORD222" + random)
-            .setSourceSubscription(new SourceSubscription("SUB234598S" + random, "master" + random));
+            .setSourceSubscription(new SourceSubscription("SUB234598S" + random, "primary" + random));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/test/TestUtil.java:573]: value change

File: src/test/java/org/candlepin/test/TestUtil.java [hunk 13]
```diff
@@ -759,7 +759,7 @@ public static Pool copyFromSub(Subscription sub) {
         }
 
         if (sub.getId() != null) {
-            pool.setSourceSubscription(new SourceSubscription(sub.getId(), "master"));
+            pool.setSourceSubscription(new SourceSubscription(sub.getId(), "primary"));
```
Comment (Ceiu) [inline src/test/java/org/candlepin/test/TestUtil.java:762]: value change

File: spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java [hunk 14]
```diff
@@ -88,7 +88,7 @@ public void shouldAutoSubscribePhysicalSystemsWithQuantity2PerSocketPair() {
             .quantity(Long.valueOf(poolQuantity))
             .subscriptionId(StringUtil.random("sourceSub-"))
             .upstreamPoolId(StringUtil.random("upstream-"))
-            .subscriptionSubKey("master"));
+            .subscriptionSubKey("primary"));
```
Comment (Ceiu) [inline spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java:91]: value change

File: spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java [hunk 15]
```diff
@@ -145,7 +145,7 @@ public void shouldAutoSubscribeGuestSystemsWithQuantity1() {
             .quantity(10L)
             .subscriptionId(StringUtil.random("sourceSub-"))
             .upstreamPoolId(StringUtil.random("upstream-"))
-            .subscriptionSubKey("master"));
+            .subscriptionSubKey("primary"));
```
Comment (Ceiu) [inline spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java:148]: value change

File: spec-tests/src/test/java/org/candlepin/spec/SubscriptionResourceSpecTest.java [hunk 16]
```diff
@@ -100,7 +100,7 @@ public void shouldSubscriptionsDerivedProvidedProductsReturnEmptyArrayWhenNull()
             .productId(prod.getId())
             .quantity(2L)
             .subscriptionId(StringUtil.random("test"))
-            .subscriptionSubKey("master");
+            .subscriptionSubKey("primary");
```
Comment (Ceiu) [inline spec-tests/src/test/java/org/candlepin/spec/SubscriptionResourceSpecTest.java:103]: value change -- omitting comment on other instances within this file

**Review by Ceiu — CHANGES_REQUESTED** [review]
Quick glance seems mostly fine, however we do not want to change the value of the subscription subkey due to the amount of legacy systems and behavioral issues that can occur if the key changes unexpectedly.

If we *really* want to remove those values from code, a better middle-ground would be to define appropriately named constants or enums with the "accepted" name, but with the existing value.

File: src/main/java/org/candlepin/liquibase/PerOrgProductsMigrationTask.java [hunk 17]
```diff
@@ -823,13 +823,13 @@ protected void migrateRelatedData() throws DatabaseException, SQLException {
...
 
-        // Migrate upstream tracking columns from subscription to master pool
+        // Migrate upstream tracking columns from subscription to primary pool
         ResultSet subscriptionInfo = this.executeQuery(
             "SELECT ss.pool_id, s.cdn_id, s.certificate_id, s.upstream_entitlement_id, " +
             "  s.upstream_consumer_id, s.upstream_pool_id " +
             "FROM cp_subscription s " +
             "JOIN cp2_pool_source_sub ss ON s.id = ss.subscription_id " +
-            "WHERE ss.subscription_sub_key = 'master'"
+            "WHERE ss.subscription_sub_key = 'primary'"
```
[the remark this answers is not in the document]
Comment (bakajstep) [reply src/main/java/org/candlepin/liquibase/PerOrgProductsMigrationTask.java]: done

File: src/main/java/org/candlepin/policy/js/pool/PoolRules.java (hunk 1 above)
Comment (bakajstep) [reply src/main/java/org/candlepin/policy/js/pool/PoolRules.java]: done

File: src/main/java/org/candlepin/policy/js/pool/PoolRules.java (hunk 2 above)
Comment (bakajstep) [reply src/main/java/org/candlepin/policy/js/pool/PoolRules.java]: done

File: src/main/java/org/candlepin/resource/OwnerResource.java [hunk 18]
```diff
@@ -1263,7 +1263,7 @@ public List<SubscriptionDTO> getOwnerSubscriptions(String ownerKey) {
         for (Pool pool : this.poolManager.listPoolsByOwner(owner).list()) {
 
             SourceSubscription srcsub = pool.getSourceSubscription();
-            if (srcsub != null && "master".equalsIgnoreCase(srcsub.getSubscriptionSubKey())) {
+            if (srcsub != null && "primary".equalsIgnoreCase(srcsub.getSubscriptionSubKey())) {
```
[the remark this answers is not in the document]
Comment (bakajstep) [reply src/main/java/org/candlepin/resource/OwnerResource.java]: done

File: src/test/java/org/candlepin/controller/EntitlementCertificateGeneratorTest.java (hunk 3 above)
Comment (bakajstep) [reply src/test/java/org/candlepin/controller/EntitlementCertificateGeneratorTest.java]: done

File: src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java (hunk 4 above)
Comment (bakajstep) [reply src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java]: done

File: src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java [hunk 19]
```diff
@@ -1347,18 +1347,18 @@ public void testRefreshPoolsWithRemovedMasterPool() {
...
-        // master pool should have been recreated
+        // primary pool should have been recreated
         pools = poolCurator.getBySubscriptionId(owner, sub.getId());
         assertEquals(2, pools.size());
-        boolean newMaster = false;
+        boolean newPrimary = false;
         for (Pool p : pools) {
-            if (p.getSourceSubscription().getSubscriptionSubKey().equals("master")) {
-                newMaster = true;
+            if (p.getSourceSubscription().getSubscriptionSubKey().equals("primary")) {
```
[the remark this answers is not in the document]
Comment (bakajstep) [reply src/test/java/org/candlepin/controller/PoolManagerFunctionalTest.java]: done

File: src/test/java/org/candlepin/controller/PoolManagerTest.java [hunk 20]
```diff
@@ -754,7 +754,7 @@ public void testRefreshPoolsDeletesOrphanedPools() {
         List<Pool> pools = new ArrayList<>();
         Product product = TestUtil.createProduct();
         Pool p = TestUtil.createPool(product);
-        p.setSourceSubscription(new SourceSubscription("112", "master"));
+        p.setSourceSubscription(new SourceSubscription("112", "primary"));
```
[the remark this answers is not in the document]
Comment (bakajstep) [reply src/test/java/org/candlepin/controller/PoolManagerTest.java]: done

File: src/test/java/org/candlepin/controller/RefresherTest.java (hunk 7 above)
Comment (bakajstep) [reply src/test/java/org/candlepin/controller/RefresherTest.java]: done

File: src/test/java/org/candlepin/model/PoolCuratorTest.java (hunk 8 above)
Comment (bakajstep) [reply src/test/java/org/candlepin/model/PoolCuratorTest.java]: done

File: src/test/java/org/candlepin/policy/AutobindRulesTest.java (hunk 9 above)
Comment (bakajstep) [reply src/test/java/org/candlepin/policy/AutobindRulesTest.java]: done

File: src/test/java/org/candlepin/resource/OwnerResourceTest.java [hunk 21]
```diff
@@ -1694,7 +1694,7 @@ public void createPool() {
     public void updatePool() {
         Product prod = this.createProduct(owner);
         Pool pool = TestUtil.createPool(owner, prod);
-        pool.setSubscriptionSubKey("master");
+        pool.setSubscriptionSubKey("primary");
```
[the remark this answers is not in the document]
Comment (bakajstep) [reply src/test/java/org/candlepin/resource/OwnerResourceTest.java]: done

File: src/test/java/org/candlepin/test/DatabaseTestFixture.java (hunk 11 above)
Comment (bakajstep) [reply src/test/java/org/candlepin/test/DatabaseTestFixture.java]: done

File: src/test/java/org/candlepin/test/TestUtil.java (hunk 12 above)
Comment (bakajstep) [reply src/test/java/org/candlepin/test/TestUtil.java]: done

File: spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java (hunk 14 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java]: done

File: spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java (hunk 15 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java]: done

File: spec-tests/src/test/java/org/candlepin/spec/SubscriptionResourceSpecTest.java (hunk 16 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/SubscriptionResourceSpecTest.java]: done

File: src/test/java/org/candlepin/controller/PoolManagerTest.java [hunk 22]
```diff
@@ -487,7 +487,7 @@ public void testRefreshPoolsOnlyRegeneratesWhenNecessary() {
 
         // Should be unchanged
         Pool p = TestUtil.createPool(product);
-        p.setSourceSubscription(new SourceSubscription(sub.getId(), "master"));
+        p.setSourceSubscription(new SourceSubscription(sub.getId(), "primary"));
```
[the remark this answers is not in the document]
Comment (bakajstep) [reply src/test/java/org/candlepin/controller/PoolManagerTest.java]: done

File: spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java (hunk 14 above)
Comment (bakajstep) [reply spec-tests/src/test/java/org/candlepin/spec/InstanceSpecTest.java]: done

**Review by Ceiu — APPROVED** [review]

**Pull request merged**